src/Chunk.ts

```diff
@@ -136,6 +136,7 @@ export default class Chunk {
 		return chunk;
 	}
 
+	augmentedHash?: string;
```

Comment:
From how this is generated, I would say this is more like a `hashAugmentation` as the actual hash is composed of more things. But maybe we can get rid of this property, cf. my other comments?

Reply:
That makes sense and will actually accomplish the lazy behaviour